File: aplicação/backend/src/main/java/eaproject/output/GetAllTypesOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.sql.Timestamp;

public class GetAllTypesOutput extends BaseOutput implements Serializable {

    @JsonProperty
    private int id;

    @JsonProperty
    private String name;

    @JsonProperty
    private int numberOfPersons;

    @JsonProperty
    private Timestamp creationDate;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getNumberOfPersons() {
        return numberOfPersons;
    }

    public void setNumberOfPersons(int numberOfPersons) {
        this.numberOfPersons = numberOfPersons;
    }

    public Timestamp getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Timestamp creationDate) {
        this.creationDate = creationDate;
    }
}

File: aplicação/frontend/ea-project/src/models/output/GetAllTypesOutput.js
import BaseOutput from './BaseOutput';

export default class GetAllTypesOutput extends BaseOutput {
    constructor(getalltypeList = [], feedbackMessages = []) {
        super(feedbackMessages);
        this.getalltypeList = getalltypeList.map(
            (getalltype) => new GetAllTypesOutput.GetAllTypeProperties(
                getalltype.id,
                getalltype.name,
                getalltype.numberOfPersons,
                getalltype.creationDate
            )
        );
    }

    static GetAllTypeProperties = class {
        constructor(id, name, numberOfPersons, creationDate) {
            this.id = id;
            this.name = name;
            this.numberOfPersons = numberOfPersons;
            this.creationDate = creationDate;
        }
    };
}
